level-2/스킬트리&49993&.js

```diff
@@ -13,4 +13,57 @@ function solution(skill, skill_trees) {
 function checkOrder(checkArr) {
     for (let i = 0; i < checkArr.length; i++) if (checkArr[i] !== i) return false
     return true
-}
\ No newline at end of file
+}
+
+//정답 2 - minkyeongJ
+function solution(skill, skill_trees) {
+	  //return할 변수
+    var answer = 0;
+
+    //파라미터로 받은 값을 모두 비교가능하게 쪼갬
+    const skillArr = Array.from(skill);
+    const skillTreesElementArr = skill_trees.map(skillTreesElement => Array.from(skillTreesElement));
+    
+    //스킬이 있는 위치 체크
+    let checkArr = [];
+    
+    /** skillTreesElementArr에 있는 배열 순서대로 skillArr의 값과 비교
+      * 일치하는 값은 skillArr의 배열번호로 변환하여 새로운 배열에 삽입
+	    * 배열을 모두 순환했는데도 값이 일치하지 않으면 27 반환
+	  */
+    skillTreesElementArr.forEach((skillTreesElement, i) => {
+        checkArr[i] = [];
+        skillArr.forEach((skillElement, j) => {
+            for(let number = 0; number < skillTreesElement.length; number++){
+                if(skillTreesElement[number] === skillElement){
+                    //같다면 number을 check[i]에 push
+                    checkArr[i].push(number);
+                    break;
+                }else if(skillTreesElement.length === number+1 && checkArr[i].length === j){
+                    //값을 다 비교했을 때 매칭되는 것이 없으면 27 입력
+                    checkArr[i].push(27);
+                }
+            }
+        })
```

Comment:
for 문 에서 if 조건에 해당하는 값을 발견하지 못했을 때 '27' 이라는 값을 push 하는 것이 목표였다면 아래와 같은 표현도 고려해보세요!
forEach 내에서 return은 다음 forEach 의 반복으로 가는 것을 의미합니다!
```suggestion
  skillArr.forEach((skillElement, j) => {
            for(let number = 0; number < skillTreesElement.length; number++){
                if(skillTreesElement[number] === skillElement){
                    checkArr[i].push(number);
                    return;
                }
            }
            checkArr[i].push(27);
        })
```